scripts/sample_dataset_request.py

```diff
@@ -0,0 +1,414 @@
+import os
+import pathlib
+from abc import ABC, abstractmethod
+from pathlib import Path
+from typing import Any
+
+import pandas as pd
+import pooch
+import xarray as xr
+from intake_esgf import ESGFCatalog
+
+OUTPUT_PATH = Path("data")
+
+
+class DataRequest(ABC):
+    """
+    Represents a request for a dataset
+
+    A polymorphic association is used to capture the different types of datasets as each
+    dataset type may have different metadata fields and may need to be handled
+    differently to generate the sample data.
+    """
+
+    def __init__(self, remove_ensembles, time_span):
+        self.remove_ensembles = remove_ensembles
+        self.time_span = time_span
+
+    @abstractmethod
+    def decimate_dataset(dataset: xr.Dataset, time_span: tuple[str, str] | None) -> xr.Dataset | None:
+        """Downscale the dataset to a smaller size."""
+        pass
+
+    def create_out_filename(self, metadata: pd.Series, ds: xr.Dataset, ds_filename: str) -> pathlib.Path:
+        """Create the output filename for the dataset."""
+        pass
+
+
+class CMIP6Request(DataRequest):
+    """
+    Represents a CMIP6 dataset request
+
+    """
+
+    def __init__(self, facets: dict[str, Any], remove_ensembles: bool, time_span: tuple[str, str] | None):
+        self.avail_facets = [
+            "mip_era",
+            "activity_drs",
+            "institution_id",
+            "source_id",
+            "experiment_id",
+            "member_id",
+            "table_id",
+            "variable_id",
+            "grid_label",
+            "version",
+            "data_node",
+        ]
+
+        self.facets = facets
+
+        self.remove_ensembles = remove_ensembles
+        self.time_span = time_span
```

Comment:
This should be replaced with a call to the initialiser
```suggestion
        super().__init__(remove_ensembles, time_span)
```